pkg/controller/controller.go

```diff
@@ -285,64 +289,64 @@ func (c *controller) getServiceClassPlanAndServiceBroker(instance *v1alpha1.Serv
 // done to validate service plan, service class exist, and handles creating
 // a brokerclient to use for a given ServiceInstance.
-func (c *controller) getServiceClassPlanAndServiceBrokerForServiceInstanceCredential(instance *v1alpha1.ServiceInstance, binding *v1alpha1.ServiceInstanceCredential) (*v1alpha1.ServiceClass, *v1alpha1.ServicePlan, string, osb.Client, error) {
+func (c *controller) getServiceClassPlanAndServiceBrokerForServiceInstanceCredential(instance *v1alpha1.ServiceInstance, bindingToUpdate *v1alpha1.ServiceInstanceCredential) (*v1alpha1.ServiceClass, *v1alpha1.ServicePlan, string, osb.Client, error) {
 	serviceClass, err := c.serviceClassLister.Get(instance.Spec.ServiceClassName)
 	if err != nil {
-		s := fmt.Sprintf("ServiceInstanceCredential \"%s/%s\" references a non-existent ServiceClass %q", binding.Namespace, binding.Name, instance.Spec.ServiceClassName)
+		s := fmt.Sprintf("ServiceInstanceCredential \"%s/%s\" references a non-existent ServiceClass %q", bindingToUpdate.Namespace, bindingToUpdate.Name, instance.Spec.ServiceClassName)
 		glog.Warning(s)
-		c.updateServiceInstanceCredentialCondition(
-			binding,
+		c.setServiceInstanceCredentialCondition(
+			bindingToUpdate,
 			v1alpha1.ServiceInstanceCredentialConditionReady,
 			v1alpha1.ConditionFalse,
 			errorNonexistentServiceClassReason,
 			"The binding references a ServiceClass that does not exist. "+s,
 		)
-		c.recorder.Event(binding, api.EventTypeWarning, "ReferencesNonexistentServiceClass", s)
+		c.recorder.Event(bindingToUpdate, api.EventTypeWarning, "ReferencesNonexistentServiceClass", s)
 		return nil, nil, "", nil, err
 	}
 
 	servicePlan := findServicePlan(instance.Spec.PlanName, serviceClass.Plans)
 	if servicePlan == nil {
 		s := fmt.Sprintf("ServiceInstance \"%s/%s\" references a non-existent ServicePlan %q on ServiceClass %q", instance.Namespace, instance.Name, instance.Spec.PlanName, serviceClass.Name)
 		glog.Warning(s)
-		c.updateServiceInstanceCredentialCondition(
-			binding,
+		c.setServiceInstanceCredentialCondition(
+			bindingToUpdate,
 			v1alpha1.ServiceInstanceCredentialConditionReady,
 			v1alpha1.ConditionFalse,
 			errorNonexistentServicePlanReason,
 			"The ServiceInstanceCredential references an ServiceInstance which references ServicePlan that does not exist. "+s,
 		)
-		c.recorder.Event(binding, api.EventTypeWarning, errorNonexistentServicePlanReason, s)
+		c.recorder.Event(bindingToUpdate, api.EventTypeWarning, errorNonexistentServicePlanReason, s)
 		return nil, nil, "", nil, fmt.Errorf(s)
 	}
 
 	broker, err := c.brokerLister.Get(serviceClass.ServiceBrokerName)
 	if err != nil {
-		s := fmt.Sprintf("ServiceInstanceCredential \"%s/%s\" references a non-existent ServiceBroker %q", binding.Namespace, binding.Name, serviceClass.ServiceBrokerName)
+		s := fmt.Sprintf("ServiceInstanceCredential \"%s/%s\" references a non-existent ServiceBroker %q", bindingToUpdate.Namespace, bindingToUpdate.Name, serviceClass.ServiceBrokerName)
```

Comment:
why change to bindingToUpdate here?

Reply:
I changed the name from `binding` to `bindingToUpdate` to indicate that the binding object is the one actually being updated instead of a clone of the binding.

Reply:
I hate be nitpicky but I do think the name changes to this and the instance method should come back out.  I totally get the temptation to rename but I don't think it adds alot (document in the godoc that the caller should pass a clone that can be updated) and this is already a very large PR to review.